### ISSUE TITLE
Can't get this to work

### ISSUE THREAD

**Issue by Thyerhazard** [naztronaut/FCW#4]
Hey! I want to start this off with me being a complete newbie. This is the first time trying to do anything like this, this is the first time I've used puTTY or done anything with SSH. So I doubt there is an issue with your work, I just need someone to hold my hand a bit if thats possible? and hopefully anyone as dumb and new as me can read your resonse and also get some useful info from it.

Ok first things first, Running a gen 1 Raspbery pi Zero W, Have 32bit Raspberry pi LITE (Formerly raspian) installed, and the pi is set up and configered for SSH with ssh file and is configured for wifi etc (And its connected to my wifi and appears to be working fine) (I will also note for this i have formatted the SD card and am starting completely blank)

So now I use puTTY and connect to the pi via its IP address, log in as pi and begin following the Readme.md 
![image](https://user-images.githubusercontent.com/102141136/159434798-9da15b0f-8310-446b-98af-1e3d02fed49d.png)

And now I will go through the readme as it says, hopefully I'm missing something obvious and its easy to fix.
First I install Git;
![image](https://user-images.githubusercontent.com/102141136/159435264-1e55960c-ab55-4e7c-a0cb-6086bcad81db.png)

Now I clone the repo;
![image](https://user-images.githubusercontent.com/102141136/159435457-73a29d03-6e6a-46be-8ead-3648e90b4933.png)

Now I head into my install directory;
![image](https://user-images.githubusercontent.com/102141136/159435592-9f8c6c28-34d3-453f-b24e-05bfa0520541.png)

Now I use the sudo python3 install.py command;
![image](https://user-images.githubusercontent.com/102141136/159436295-afbf97d7-8182-4fef-9839-f44d264051ad.png)

I also scrolled up through all the text that was generated and noticed these two failed to fetch down the bottom (Unsure if this is relavent or not) 
![image](https://user-images.githubusercontent.com/102141136/159436632-6fc968bd-91f8-49ff-a9dc-8420e5d92995.png)

At this point the guide says to change the LED_COUNT for Config.py but not *HOW* to do that, I understand this might be something very simple for %99 of people but for someone like me who has barely used a terminal and is trying to learn this can be frustrating, I tried for a while assumeing that its like the sudo nano part later maybe like sudo nano config.py? I don't know I couldn't figure it out.

Either way I continued going to make sure i could get it working before working out the smaller details; 

Now I get to the part that tells me to Install and Run Flask, so I run that.
![image](https://user-images.githubusercontent.com/102141136/159437408-a5e705b1-22e2-4208-8048-a01ae14496b6.png)

Now I do the next step which is 
*cd /etc/apache2/sites-available
sudo nano fcw.conf*
![image](https://user-images.githubusercontent.com/102141136/159437665-8da0f086-f50a-4c76-b2d2-ab8e40492185.png)
![image](https://user-images.githubusercontent.com/102141136/159437740-8b09c13c-6c27-4b00-a863-5c6b2faeae2c.png)

I was stuck here for a while before i figured out how to use this file editor a bit, So i used Cntl + O to save, and then Cntl + x to leave.

Then i disable the defualt config and add the new one and restart apache;
![image](https://user-images.githubusercontent.com/102141136/159438213-fa482efe-af32-4923-9ebd-74ba35c40d31.png)

I see this text;
(To activate the new configuration, you need to run:
systemctl reload apache2
Site 000-default disabled.
To activate the new configuration, you need to run:
systemctl reload apache2)

I'm unsure what this means but its not mentioned in the guide so for now I won't do anything, but if it does not work maybe I'll try doing 
(systemctl reload apache2) see what that does.

Last things last, guide says to visit (http://ip_addr/fcw and it sould be work)

But I just get this page
![image](https://user-images.githubusercontent.com/102141136/159439225-fa375959-8751-476d-a5b1-2b8df96a3399.png)

If i just visit the Pis IP address i get a page that says (APACHE 2 DEBIAN DEFAULT PAGE *IT WORKS*)

In conclusion I appreciate you work and completely understand if you don't have the time to look through this for me, But if you get the time I would appriciate it so much if you could tell me what I've done wrong so i can hook up same LEDs and make funny colours.

(note I have tried exactly what I've done about 5 times every time on a fresh format SD card, and also tried a diffrent SD card and even another Zero W) 

thanks again man! this is my first time on GitHub so I hope the way I have presented this is acceptible :D

**Comment by Thyerhazard** [comment]
Update, I tired the systemctl reload apache2 it asked me to enter password etc and looked like it did something, but I get the same NOT FOUND page so no clue :(

**Comment by Thyerhazard** [comment]
Also note, I've also followed a few of you other projects but nothing seems to work :( just incase I'm in the wrong spot I'm trying to control indiviudble addressible RGB leds over wifi for colour, brightness and animation control.

**Comment by naztronaut** [comment]
You should put the project files in /var/www/html first. Apache only has access to this directory. I think I skipped that in my Readme because it's a short project and I've covered it in a lot of my other tutorials, I'll try and update that at some point. 

Second thing I noticed is that the virtual environment never got created. It tells you to run `sudo apt-get install python3-venv` in one of the errors. Which is strange because your first screenshot showed that it was installed (something else failed though). 

Third thing I noticed is that one of the errors is mentioning my other project, the RGBW Controller but this repo deals with Ws2812b leds which are pretty different. Which project are you working on? The RGBW one is definitely easier but I also build up to that with 4 tutorials that came before that:

1. Running Apache: https://www.easyprogramming.net/raspberrypi/pi_apache_web_server.php
2. Running Flask: https://www.easyprogramming.net/raspberrypi/pi_flask_app_server.php
3. Running Flask behind apache: https://www.easyprogramming.net/raspberrypi/pi_flask_apache.php
4. Controlling LED from your browser: https://www.easyprogramming.net/raspberrypi/browser_control_led.php
5. Controlling RGBW Leds: https://www.easyprogramming.net/raspberrypi/rgbw_led_strip.php

My goal was to build on previous knowledge. If you don't know how to work with apache/flask separately, I'd highly recommend starting there. They are quick and can help you understand a lot of the different pieces of the RGBW and this (FCW) project and it's easier to spot where things go wrong. 

Last question, what version of Raspberry Pi OS are you using? It's still referencing 'apt get' which I thought they stopped doing a while ago. A fresh version of Raspberry Pi OS could also be good idea to get rid of any bugs/breaks that could exist (assuming you don't have any other projects running on  your Pi right now). 

Good luck! I'm here for questions but I may be slow at responding from time to time.

**Comment by Thyerhazard** [comment]
I'm was doing the Ws2812b one.

I'm thinking I'll go with an ESP32 for this project anyway at this point, I apprciate what you're trying to do but jumping back and fourth between 5 or more completely different tutorials is just confusing to someone like me who has very little experience and only really cares about having my lights make pretty colours haha, 

If you actually wanted to learn the nitty gritty in depth way to do this stuff I get it, but for someone who just wants to hook up a Ws2812b strip and throw it under the desk its a little confusing.

**Comment by naztronaut** [comment]
Check out Adafruit's tutorials, they have some good ones where you can just use their libraries and plug n' play WS2812b leds.  

My goal was to teach the nitty gritty.  It's also not easy to easy to create something that can be plug and play like this because there are too many components and some require manual intervention, such as turning on the virtual environment and connecting the flask app to apache.  So you need the extra steps. For that reason, I'd label this project as somewhat advanced. 

There are plenty of other repos out there where you can plug in the lights and run a python script, that could be enough for you. If you know some python, you can actually just take the led_watchdog.py in this project, add a way to send arguments to it, and run it in command line without the rest of the bells and whistles. So the end result would be something like:

`sudo python3 led_watchdog.py color_wipe`

And that would do the color_wipe animation.  More examples also available on the rpi-ws281x library: https://github.com/rpi-ws281x/rpi-ws281x-python/blob/master/examples/strandtest.py

Good luck!

**Comment by Thyerhazard** [comment]
I'll look into it, cheers my dude.

**Comment by rubyone** [comment]
@Thyerhazard 

I was also trying out the project and found the mistake.
The Apache config fcw.conf at WSGIScriptAlias is searching for file "**fcw**.wsgi" which is not existent with that name in the project.
`WSGIScriptAlias /fcw  /var/www/html/fcw/fcw.wsgi`
 rename to:         
`WSGIScriptAlias /fcw  /var/www/html/fcw/led.wsgi`

hope that helps
